server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoGridAggregator.java

```diff
@@ -79,7 +86,8 @@ public void collect(int doc, long bucket) throws IOException {
 
                     long previous = Long.MAX_VALUE;
                     for (int i = 0; i < valuesCount; ++i) {
-                        final long val = values.nextValue();
+                        final GeoPoint point = values.nextValue();
+                        final long val = longEncoder.encode(point.getLon(), point.getLat(), precision);
```

Comment:
the main reason for this change is so that more than one bucket can be added to `bucketOrds` per doc-value.